packages/app/src/components/inspector/SymbolInstanceOverridesRow.tsx

```diff
@@ -0,0 +1,336 @@
+  Button,
+} from 'noya-designsystem';
+import { ApplicationState, Layers, Overrides } from 'noya-state';
+import {
+  getSharedStyles,
+  getSharedTextStyles,
+  getSymbols,
+} from 'noya-state/src/selectors/themeSelectors';
+import { memo, ReactNode, useCallback, useMemo } from 'react';
+import { useApplicationState } from '../../contexts/ApplicationStateContext';
+import * as InspectorPrimitives from './InspectorPrimitives';
+
+const NO_STYLE = 'none';
+
+interface Props {
+  canOverride: boolean;
+  symbolMaster: Sketch.SymbolMaster;
+  overrideValues: Sketch.OverrideValue[];
+  onSetOverrideValue: (overrideName: string, value: string) => void;
+  onResetOverrideValue: () => void;
+}
+
+type Selectors = {
+  symbols: Sketch.SymbolMaster[];
+  textStyles: Sketch.SharedStyle[];
+  themeStyles: Sketch.SharedStyle[];
+};
+
+const ThemeStyleSelector = ({
+  themeStyles,
+  sharedStyleID,
+  onChange,
+}: {
+  themeStyles: Sketch.SharedStyle[];
+  sharedStyleID: string;
+  onChange: (value: string) => void;
+}) => {
+  const themeStylesOptions = useMemo(
+    () => [NO_STYLE, ...themeStyles.map((style) => style.do_objectID)],
+    [themeStyles],
+  );
+
+  const getThemeStylesTitle = useCallback(
+    (id) =>
+      id === NO_STYLE
+        ? 'No Layer Style'
+        : themeStyles.find((style) => style.do_objectID === id)!.name,
+    [themeStyles],
+  );
+
+  return (
+    <Select
+      id="theme-style-selector"
+      value={sharedStyleID}
+      options={themeStylesOptions}
+      getTitle={getThemeStylesTitle}
+      onChange={onChange}
+    />
+  );
+};
+
+const TextStyleSelector = ({
+  textStyles,
+  sharedStyleID,
+  onChange,
+}: {
+  textStyles: Sketch.SharedStyle[];
+  sharedStyleID: string;
+  onChange: (value: string) => void;
+}) => {
+  const textStylesOptions = useMemo(
+    () => [NO_STYLE, ...textStyles.map((style) => style.do_objectID)],
+    [textStyles],
+  );
+
+  const getTextStylesTitle = useCallback(
+    (id) =>
+      id === NO_STYLE
+        ? 'No Layer Style'
+        : textStyles.find((style) => style.do_objectID === id)!.name,
+    [textStyles],
+  );
+  return (
+    <Select
+      id="text-style-selector"
+      value={sharedStyleID}
+      options={textStylesOptions}
+      getTitle={getTextStylesTitle}
+      onChange={onChange}
+    />
+  );
+};
+
+const SymbolMasterSelector = ({
+  symbols,
+  symbolId,
+  onChange,
+}: {
+  symbols: Sketch.SymbolMaster[];
+  symbolId: string;
+  onChange: (value: string) => void;
+}) => {
+  const symbolSourceOptions = useMemo(
+    () => symbols.map((symbol) => symbol.symbolID),
+    [symbols],
+  );
+
+  const getSymbolMasterTitle = useCallback(
+    (id) => symbols.find((symbol) => symbol.symbolID === id)!.name,
+    [symbols],
+  );
+  return (
+    <Select
+      id="symbol-instance-source"
+      value={symbolId}
+      options={symbolSourceOptions}
+      getTitle={getSymbolMasterTitle}
+      onChange={onChange}
+    />
+  );
+};
+
+function getOverrideElements(
```

Comment:
Since we're already passing in the entire `state` object, I don't think we need to pass `selectors`. Instead, we can  call these first thing within `getOverrideElements`:

```ts
  const themeStyles = getSharedStyles(state);
  const textStyles = getSharedTextStyles(state);
  const symbols = getSymbols(state);
```

Reply:
Isn't it better to only call it once? Instead of calling it every time the getOverrideElements happens 🤔 Oh it doesn't matter?

Reply:
It is better in terms of perf, but for most function calls (especially the first 2 here) it's going to be extremely minor. I'd go with the simpler code and wait till we actually see it causing issues before optimizing here